flaskbb/utils/requirements.py

```diff
@@ -168,7 +168,11 @@ def _get_forum_from_request(self):
 def can_user_access_forum(user, forum):
     user_groups = {g.id for g in user.groups}
     forum_groups = {g.id for g in forum.groups}
-    return bool(forum_groups & user_groups)
+    if forum.subforum_parent_id:
+        parent_access = can_user_access_forum(user, Forum.query.filter_by(id=forum.subforum_parent_id).first())
+        return bool(forum_groups & user_groups) and parent_access
+    else:
+        return bool(forum_groups & user_groups)
```

Comment:
`bool(forum_groups & user_groups)` дублируется, не хорошо. Лучше так:

```suggestion
    if forum.subforum_parent_id and not can_user_access_forum(user, Forum.query.filter_by(id=forum.subforum_parent_id).first():
        return False
    return bool(forum_groups & user_groups)
```